### ISSUE TITLE
custom wrapper for elements

### ISSUE THREAD

**Issue by mateusarbex** [meliorence/react-native-render-html#429]

**Comment by jsamr** [comment]
@mateusarbex Would be very much appreciated if you provided a minimal description. What I understand is basically having the option to render the default renderer in a custom renderer, is that it?

**Comment by mateusarbex** [comment]
Sorry for the minimal description but what I want is a way to use the the parsed html element with a custom wrapper renderer. Since the renderer prop happens before the html is parsed, I can't wrapp the result in a wrapper I designed. For example: I want to wrapp the parsed <img> inside a <TouchableOpacity>

**Comment by jsamr** [comment]
Would that work out?
```jsx
const renderers = {
  img: ({ DefaultRenderer, ...props }) => {
    return (
      <TouchableOpacity onPress={/* do stuff */}>
        <DefaultRenderer {...props} />
      </TouchableOpacity>
    );
  },
};
```

**Comment by mateusarbex** [comment]
Yes, it would work perfectly!

**Comment by jsamr** [comment]
@mateusarbex Great. I'll design the next foundry release "renderers" prop with this feature in mind. I'm working on it, expect a first pre-release in a few days ;-)

**Comment by mateusarbex** [comment]
Thanks!

**Comment by ymj** [comment]
> @mateusarbex Would be very much appreciated if you provided a minimal description. What I understand is basically having the option to render the default renderer in a custom renderer, is that it?

That option would be highly appreciated. I would love to have the option to use a custom renderer for elements with certain attributes, and defer to the default rendered if those attributes are not present.

**Comment by jsamr** [comment]
This feature is available in the foundry pre-release (see #430 for instruction on how to try it out). You can use `TDefaultRenderer` prop for that purpose. Also, note that `TDefaultRenderer` will always support `onPress` prop, whether it renders a `Text` or `View`. See for example this snippet: https://github.com/meliorence/react-native-render-html/blob/682a9b91f21b39b63561cf69670cbf5e8142aaf9/demo/snippets/customRenderers.tsx#L121

**Comment by jsamr** [comment]
Closing now since the stable beta which fixes this issue [has just been released](https://github.com/meliorence/react-native-render-html/releases/tag/v6.0.0-beta.0).
